src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,93 @@
+package codeu.controller;
+
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import org.mindrot.jbcrypt.BCrypt;
+
+/** Servlet class responsible for the Admin page. */
+public class AdminServlet extends HttpServlet {
+
+  private UserStore UserStore;
+  private ConversationStore ConversationStore;
+  private MessageStore MessageStore;
+
+  /**
+   * Set up state for handling login-related requests. This method is only called when running in a
+   * server, not when running in a test.
+   */
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore UserStore) {
+    this.UserStore = UserStore;
+  }
+
+  void setConversationStore(ConversationStore ConversationStore) {
+    this.ConversationStore = ConversationStore;
+  }
+
+  void setMessageStore(MessageStore MessageStore) {
+    this.MessageStore = MessageStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+      String username = (String)request.getSession().getAttribute("user");
+      if (username == null) {
+        // user is not logged in, don't let them create a conversation
```

Comment:
is creating a conversation what they would've been able to do if it weren't null?

Reply:
Same question. How is the admin page related to conversations?